### PR TITLE
Feature/native price service

### PR DESCRIPTION
**Pull request by DevAseel**
**Summary of change:**

- Created sync record fetcher class.
- Created all pairs finder class.
- Created native price calculator class.
- Created native price saver class.
- Created native price consumer class.

**Steps for testing**
- Make sure your local git is up to date
```bash
git pull
```

- Checkout the branch `feature/native-price-service`
```bash
git checkout feature/native-price-service
```

- Run the test across the test-net
```bash
npm run test
```

- Confirm tests have passed.
- Run example.

### CODE REVIEW

File: src/services/consumers/NativePriceConsumer.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    this.nativePriceSaver = nativePriceSaver;
+  }
+
+  /**
+   * Native Price consumer
+   * @param blockNumber the height of the block
+   */
+  async apply(blockNumber: number): Promise<NativePriceConsumerResult> {
+    // find pairSyncEvent
+    const pairSyncEventFinderResult = await this.pairSyncEventFinder.apply(blockNumber);
```
Comment (jjfluid) [inline src/services/consumers/NativePriceConsumer.ts:51]: definitely want to be able to specify the pair addresses that will be used for pricing.
for example if we provide 3 pair addresses, for: USDC_WCSPR, USDT_WCSPR, DAI_WCSPR, we will need to do this:
 - for each pair:
   - get the latest sync
   - price the assets in the pair relative to eachother using the reserves
   - determine which token in the pair is WCSPR
   - return the following values:
     - the price of WCSPR in the units of the USD stablecoin
     - the amount of the USD stablecoin in the reserves
 - then we take an average of all the WCSPR price's which will be weighted by the amount of USD stablecoin in the reserves of each
 - this is now the price that gets saved for the block in the NativePricesUsd entity

Comment (DevAseel) [reply src/services/consumers/NativePriceConsumer.ts:51]: Currently I can't find a way to determine the type of token (USDC_WCSPR, USDT_WCSPR, DAI_WCSPR)

Comment (jjfluid) [reply src/services/consumers/NativePriceConsumer.ts:51]: give me an example to lookup

Comment (DevAseel) [reply src/services/consumers/NativePriceConsumer.ts:51]: Not an issue anymore, this is related to what we have discussed on Tuesday's meeting. All is good now.

**Review by jjfluid — APPROVED** [review]

**Comment by DevAseel** [comment]
Closing this one out as it was included in this [PR](https://github.com/fluidefi/fluidefi-caspernet-aggregator-tools/pull/64).

**Pull request closed without merging**